File: c++/basics/2c_reverse_ll.cpp
//Reverse the LL.

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {  
        if(!head || !head->next) return head;   
        ListNode* fwd = NULL;
        ListNode* bwd = NULL;
        ListNode* temp = head;
        while(temp){
            fwd = temp->next;
            temp->next = bwd;
            bwd = temp;
            temp = fwd;
        }
        return bwd;
    }
}; 

File: js/basics/2c_reverse_ll.js
//Reverse the LL.

/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    if(!head || !head.next) return head
    let curr = head
    let prev = null, fwd = null 
    while(curr){
        fwd = curr.next
        curr.next = prev 
        prev = curr
        curr = fwd
    }
    return prev
};